src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/CodeGeneration/MethodGenerator.cs

```diff
@@ -137,6 +140,27 @@ private static MethodDeclarationSyntax GenerateMethodDeclarationWorker(
         return AddFormatterAndCodeGeneratorAnnotationsTo(methodDeclaration);
     }
 
+    private static IParameterSymbol FilterAttributes(IParameterSymbol parameter, CodeGenerationDestination destination)
+        => parameter.WithAttributes(parameter.GetAttributes().WhereAsArray(static (a, destination) => FilterAttribute(a, destination), destination));
+
+    private static bool FilterAttribute(AttributeData attribute, CodeGenerationDestination destination)
+    {
+        if (destination is CodeGenerationDestination.InterfaceType)
+        {
+            // EnumeratorCancellation serves no purpose in an interface.  Filter it out.
+            return attribute.AttributeClass is not
+            {
+                Name: nameof(EnumeratorCancellationAttribute),
+                ContainingNamespace.Name: nameof(System.Runtime.CompilerServices),
+                ContainingNamespace.ContainingNamespace.Name: nameof(System.Runtime),
+                ContainingNamespace.ContainingNamespace.ContainingNamespace.Name: nameof(System),
+                ContainingNamespace.ContainingNamespace.ContainingNamespace.ContainingNamespace.IsGlobalNamespace: true,
```

Comment:
>ContainingNamespace.ContainingNamespace.ContainingNamespace.ContainingNamespace [](http://example.com/codeflow?start=16&length=79)

I'm pretty sure I asked this a long time ago, but what's the reason for the checking every level?

Reply:
just to be precise that this is indeed the type we're looking for.